ecl/hql/hqlgram2.cpp

```diff
@@ -1688,13 +1688,15 @@ void HqlGram::addAssignment(attribute & target, attribute &source)
         {
             case type_record:
             case type_row:
+            case type_null: // Assign no_null
```

Comment:
Why is this allowed?  When would the source be type_null?

Reply:
When it is a no_null.  I.e.  [ ]